executor/src/witgen/jit/processor.rs

```diff
@@ -411,48 +465,94 @@ impl<'a, T: FieldElement> Processor<'a, T> {
         }
     }
 
-    /// Returns true if all polynomial identities are solved for at least `self.block_size` rows
-    /// in the middle of the rows. A polynomial identities is solved if it evaluates to
-    /// a known value.
-    fn all_polynomial_identities_solved_in_block<FixedEval: FixedEvaluator<T>>(
-        &self,
-        witgen: &WitgenInference<'a, T, FixedEval>,
-    ) -> bool {
+    /// Returns all pairs of polynomial identity and row where the identity is not solved
+    /// in `self.block_size` contiguous rows. A polynomial identities is considered solved if
+    /// it evaluates to a known value.
+    /// If a polynomial identity is solved for `self.block_size` contiguous rows, it is not
+    /// returned, not even on the rows where it is not solved.
+    fn unsolved_polynomial_identities_in_block<'b, FixedEval: FixedEvaluator<T>>(
+        &'b self,
+        witgen: &'b WitgenInference<'a, T, FixedEval>,
+    ) -> impl Iterator<Item = (&'a Identity<T>, i32)> + 'b {
```

Comment:
We could return `PolynomialIdentity` instead of `Identity` here (the only time it's used we immediately unpack).